lib/remote_node.go

```diff
@@ -250,6 +271,10 @@ func (rn *RemoteNode) IsTerminated() bool {
 	return rn.connectionStatus == RemoteNodeStatus_Terminated
 }
 
+func (rn *RemoteNode) GetRemoteNodeStatus() RemoteNodeStatus {
```

Comment:
nit: it's clear based on the receiver type that it's a remote node status
```suggestion
func (rn *RemoteNode) GetStatus() RemoteNodeStatus {
```